yellow-paper/docs/public-vm/avm.md

```diff
@@ -302,9 +307,42 @@ nestedMachineState = MachineState {
     l2GasLeft: callingContext.machineState.memory[instr.args.gasOffset+1],
     pc: 0,
     memory: uninitialized,
+    internalCallStack: empty,
 }
 ```
+> Note: the sub-context machine state's `l*GasLeft` is initialized based on the call instruction's `gasOffset` argument. The caller allocates some amount of L1 and L2 gas to the nested call. It does so using the instruction's `gasOffset` argument. In particular, prior to the message call instruction, the caller populates `M[gasOffset]` with the sub-context's initial `l1GasLeft`. Likewise it populates `M[gasOffset+1]` with `l2GasLeft`.
+
 > Note: recall that `INITIAL_MESSAGE_CALL_RESULTS` is the same initial value used during [context initialization for a public execution request's initial message call](#context-initialization-for-initial-call).
 > `STATICCALL_OP` and `DELEGATECALL_OP` refer to the 8-bit opcode values for the `STATICCALL` and `DELEGATECALL` instructions respectively.
 
-### Updating the calling context after nested call halts
\ No newline at end of file
+### Updating the calling context after nested call halts
+When a message call's execution encounters an instruction that itself triggers a message call, the nested call executes until it reaches a halt. At that point, control returns to the caller, and the calling context is updated based on the sub-context and the message call instruction's transition function. The components of that transition function are defined below.
+
+The success or failure of the nested call is captured into memory at the offset specified by the call instruction's `successOffset` input:
+```
+context.machineState.memory[instr.args.successOffset] = subContext.results.reverted
```

Comment:
If this boolean is true, we should interpret it  as "reverted", i.e., failure? Otherwise, we might rename this boolean

Reply:
Good catch! Should be `success = !reverted`